core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java

```diff
@@ -152,9 +163,35 @@ public Stack convert(StackV4Request source) {
         if (source.getImage() != null) {
             stack.getComponents().add(getImageComponent(source, stack));
         }
+        extendGatewaySecurityGroupWithDefaultGatewayCidrs(stack);
         return stack;
     }
 
+    private void extendGatewaySecurityGroupWithDefaultGatewayCidrs(Stack stack) {
+        Set<InstanceGroup> gateways =
+                stack.getInstanceGroups().stream().filter(ig -> InstanceGroupType.isGateway(ig.getInstanceGroupType())).collect(Collectors.toSet());
+        Set<String> defaultGatewayCidrs = defaultGatewayCidr
+                .stream()
+                .filter(StringUtils::isNotBlank)
+                .collect(Collectors.toSet());
+        for (InstanceGroup gateway : gateways) {
+            Set<SecurityRule> rules = gateway.getSecurityGroup().getSecurityRules();
+            if (!defaultGatewayCidrs.isEmpty()) {
```

Comment:
I suggest pulling the `if` outside the `for` loop as the condition is constant throughout all iterations.